cellarium/cas/data_preparation/callbacks.py

```diff
@@ -0,0 +1,50 @@
+import typing as t
+
+import numpy as np
+
+if t.TYPE_CHECKING:
+    import anndata
+
+TOTAL_MRNA_UMIS_COLUMN_NAME = "total_mrna_umis"
+
+
+def calculate_total_mrna_umis(adata: "anndata.AnnData") -> "anndata.AnnData":
+    """
+    Calculate the total mRNA UMIs (Unique Molecular Identifiers) for each observation in the AnnData object and add them
+    as a new column in the ``.obs`` attribute. It is recommended to use this callback before data sanitization to
+    calculate all expressed genes.
+
+    :param adata: The annotated data matrix from which to calculate the total mRNA UMIs.
+
+    :return: A new AnnData object with the same data as ``adata`` but with an additional field in ``.obs`` containing
+        the total mRNA UMIs for each observation.
+
+    .. note::
+       The function assumes that the ``.X`` attribute of the input AnnData is a count matrix where each element i,j
+       represents the count of unique transcripts (UMIs) for the ith observation and jth variable (gene).
+
+    Example:
+    ________
+        >>> adata_with_umis = calculate_total_mrna_umis(adata=adata)
+        >>> 'total_mrna_umis' in adata_with_umis.obs.columns
+        True.
+    """
+    adata_r = adata.copy()
```

Comment:
Let's do this inline.

Reply:
Also, mention in the docstring that the operation is inline and modifies the input adata file